app/controllers/api/tags_controller.rb

```diff
@@ -0,0 +1,87 @@
+module Api
+  # Api controller for tags
+  # Uses Rails' RESTful routes (check 'rake routes' for the configured routes)
+  class TagsController < MainApiController
+    DEFAULT_FIELDS = [:id, :name, :description, :creator, :use].freeze
+
+    def index
+      @assignment = Assignment.find_by(id: params[:assignment_id])
+
+      respond_to do |format|
+        parent = @assignment || current_course
+        tags = parent.tags.includes(:role, :groupings).order(:name)
+
+        tag_info = tags.map do |tag|
+          {
+            id: tag.id,
+            name: tag.name,
+            description: tag.description,
+            creator: tag.role.display_name,
+            use: tag.groupings.size
+          }
+        end
+        format.xml do
+          render xml: tag_info
+        end
+        format.json do
+          render json: tag_info
+        end
+      end
+    end
+
+    # Creates a new instance of the tag.
+    def create
+      params.permit(:name, :description)
+      begin
+        assignment = current_course.assignments.find_by(id: params[:assignment_id])
+        new_tag = Tag.new(course: current_course, role: current_role, assessment: assignment, **self.tag_params)
+        if new_tag.save && params[:grouping_id]
```

Comment:
I see, so here we want to raise an error if the tag is invalid (e.g. if the tag name/description is omitted). So you can turn the previous line into `Tag.create!`, which will save the object (so no need for the `new_tag.save` here) raise an error if any validations fail.

Make sure to add tests for this as well.